src/core/runtime-field.js

```diff
@@ -0,0 +1,71 @@
+'use strict';
+
+const isNil = require('lodash.isnil');
+const validType = [
+    'boolean',
+    'composite',
+    'date',
+    'double',
+    'geo_point',
+    'ip',
+    'keyword',
+    'long',
+    'lookup'
+];
+
+class RuntimeField {
```

Comment:
Could you please add doc comments for the class and it's methods? You can check any other file for reference.

Reply:
Also, please add a note here mentioning that this feature was added to Elasticsearch in v7.11?